tools/boilerplate.py

```diff
@@ -49,19 +49,7 @@
 @_autogen_docstring(Axes.%(func)s)
 def %(func)s(%(argspec)s):
     %(ax)s = gca()
-    # Deprecated: allow callers to override the hold state
-    # by passing hold=True|False
-    %(washold)s = %(ax)s._hold
-%(sethold)s
-    if hold is not None:
-        %(ax)s._hold = hold
-        from matplotlib.cbook import mplDeprecation
-        warnings.warn("The 'hold' keyword argument is deprecated since 2.0.",
-                      mplDeprecation)
-    try:
-        %(ret)s = %(ax)s.%(func)s(%(call)s)
-    finally:
-        %(ax)s._hold = %(washold)s
+    %(ret)s = %(ax)s.%(func)s(%(call)s)
```

Comment:
Even simpler would be to just do `return gca().func(...)`, which avoids the need for temporary variables and thus allows you to get rid of the "gensym-like facility" mentioned lower in this script.

Reply:
:+1: the only reason we needed the local reference to the return value was so we could reset the hold.